### PR TITLE
Fix CMake warnings from jit/CMakeLists.txt

### PR DESCRIPTION
**Pull request by Gnbrkm41**
* https://cmake.org/cmake/help/latest/command/if.html#command:if - Per legacy, the else() and endif() commands admit an optional <condition> argument. If used, it must be a verbatim repeat of the argument of the opening if command.

During the JIT build, currently the CMake generates a warning for these particular lines of code:
https://github.com/dotnet/runtime/blob/e0ec035994179e8ebd6ccf081711ee11d4c5491b/src/coreclr/src/jit/CMakeLists.txt#L315-L325
```
  -- Checking prototype statfs for HAVE_NON_LEGACY_STATFS - True
  CMake Warning (dev) in src/jit/CMakeLists.txt:
    A logical block opening on the line

      /runtime/src/coreclr/src/jit/CMakeLists.txt:315 (if)

    closes on the line

      /runtime/src/coreclr/src/jit/CMakeLists.txt:325 (endif)

    with mis-matching arguments.
  This warning is for project developers.  Use -Wno-dev to suppress it.
```

I am not entirely sure if it's fine if we omit the argument entirely (the documents says it's optional, for legacy reasons), but looks like at least we can do this.

cc @jeffschwMSFT? Looks like a coreclr infra change to me, but please correct me if I'm wrong.

### CODE REVIEW

**Comment by jeffschwMSFT** [comment]
Thanks @Gnbrkm41 for pushing the change.  Adding some folks to the pr.

@jashook @BruceForstall

**Review by BruceForstall — APPROVED** [review]
I'm ok with this, but I'm also ok with omitting it altogether (using `endif()`) which would be simpler in this case.

File: src/coreclr/src/jit/CMakeLists.txt [hunk 1]
```diff
@@ -322,7 +322,7 @@ else()
   elseif(CLR_CMAKE_HOST_SUNOS)
```
Comment (echesakovMSFT) [inline src/coreclr/src/jit/CMakeLists.txt:322]: It is interesting that we test for target everywhere except one case - SunOS where we check for host.

Comment (jkotas) [reply src/coreclr/src/jit/CMakeLists.txt:322]: @am11 Is there a reason for this?

**Comment by jkotas** [comment]
Thanks for submitting the fix. This was fixed by a different PR that got merged already. Closing.

**Pull request closed without merging**